akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -675,6 +676,25 @@ trait FlowOps[+Out, +Mat] {
    */
   def recover[T >: Out](pf: PartialFunction[Throwable, T]): Repr[T] = via(Recover(pf))
 
+  /**
+   * RecoverTermination allows to transform any kind of termination (success or failure) into either a successful
+   * termination or a failed termination.
```

Comment:
Perhaps this advanced signature is more difficult to understand than necessary, when also taking into account the other already existing operators:

* Done => Done - doesn't make much sense
* Done => Exc - we are missing operator for this case, but is it needed?
* Exc => Exc - mapError
* Exc => Done - `recoverWith(Source.empty)` somewhat verbose syntax but not too bad

I also created issue https://github.com/akka/akka/issues/24992

Reply:
I agree, I think it was the `recoverWith(Source.empty)` use-case that wasn't apparent to me. If you look at it from this way, the current impl. is more or less equivalent to

```scala
.recoverWithRetries(-1, { case _ =>  Source.empty })
```

As you noted, we lose the possibility of Done => Exc, not sure if anybody needs that anyway.

Reply:
Thanks for the analysis! Yeah it seems we're fine if we only added the `recoverWithComplete`, would we be ok with that? Should help people discover this, rather than the Source.empty trick

Reply:
Adding `recoverWithComplete` is fine

Reply:
To clarify: The intended API will look like this:

```scala
// scaladsl
.recoverWithComplete[RuntimeException]

//javadsl
.recoverWithComplete[RuntimeException.class]
```

And it can be implemented by the means of `recoverWithRetries` like:

```scala
def recoverWithComplete[E <: Throwable] = recoverWithRetries(-1, { case _: E => Source.empty })
```

Untested code, just to outline the idea. Any other API you'd envision?

Reply:
@markusthoemmes I think the scaladsl should be a PartialFunction since all the other recover have that. For javadsl I think it's better with Class param, as discussed in #24992. 
It would be great if you can fix that ticket at the same time. In this PR, or in separate PR.

Reply:
To further narrow the proposed API in:

```scala
// scaladsl
.recoverWithComplete {
  case ex: RuntimeException => Done
  case ex: AnyOtherException => Done
}

//javadsl
.recoverWithComplete[RuntimeException.class]
.recoverWithComplete[AnyOtherException.class]
```

Is that what's in your mind @patriknw? Shall I open an issue for this to get it formerly triaged and proposed?